tiny_dnn/layers/global_average_pooling_layer.h

```diff
@@ -109,8 +109,9 @@ class global_average_pooling_layer : public layer {
       core::OpKernelConstruction(layer::device(), &params_);
 
     layer::set_backend_type(backend_type);
-    if (backend_type == backend_t::internal || backend_type == backend_t::avx ||
-        backend_type == backend_t::nnpack) {
+    if (backend_type == core::backend_t::internal ||
+        backend_type == core::backend_t::avx ||
+        backend_type == core::backend_t::nnpack) {
```

Comment:
My understanding is that we can always fallback to internal backend when selected backend doesn't support layer's operation.